iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/RatioTapChangerAdderImpl.java

```diff
@@ -144,7 +143,22 @@ public RatioTapChangerAdder setRegulating(boolean regulating) {
 
     @Override
     public RatioTapChangerAdder setTargetV(double targetV) {
-        this.targetV = targetV;
+        if (!Double.isNaN(targetV)) {
```

Comment:
I think @jeandemanged you just have to check if regulation mode is different than VOLTAGE, no?

Reply:
hm, checking for what purpose ?
if API user sets a targetV with a non-NaN value, then implicitly he wants to be in VOLTAGE regulation mode

Reply:
Because I understand that you can have a NaN targetV and have a REACTIVE_POWER mode, but set a targetV that will change the mode, no?